### DISCUSSION TITLE
NextJS 9.3 to 9.4 upgrade

### DISCUSSION THREAD

**Discussion by PJijin** [vercel/next.js#14056] [Help] [answered]
I am trying to update 9.3 to 9.4. I have updated the version on package.json  ```"next": "^9.4.0",``` 

After that, a white page is shown and this error in console. 

```
Uncaught SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at eval (platform.js?2a2e:14)
    at Object../node_modules/ally.js/util/platform.js (main.js?ts=1591861175888:1535)
    at __webpack_require__ (webpack.js?ts=1591861175888:863)
    at fn (webpack.js?ts=1591861175888:151)
    at eval (detect-focus.js?93d2:19)
    at Object../node_modules/ally.js/supports/detect-focus.js (main.js?ts=1591861175888:851)
    at __webpack_require__ (webpack.js?ts=1591861175888:863)
    at fn (webpack.js?ts=1591861175888:151)
    at eval (supports.js?750b:21)
```

Works fine on 9.3 . Any idea why this occurs?

**Comment by rafaelalmeidatk** [comment]
Hi! After the upgrade, try to delete the `node_modules` folder and run the yarn/npm install command again to restore it with the right dependencies

**Reply by PJijin** [reply]
Still that error.

**Reply by rafaelalmeidatk** [reply]
Can you try deleting the `.next` folder too?

**Reply by PJijin** [reply]
Tried. Removed .next, yarn.lock, node_modules . still error.

**Reply by ricokahler** [reply]
I also tried deleting everything (`.next`, `node_modules`, `package-lock.json`) and I still have the error.

**Comment by ricokahler** [comment]
I'm also experiencing this issue. I did a bit of digging and it seems like the issue lies within `@next/react-dev-overlay`. All I did was follow the dependencies. It's throwing in `platform.js` which is used by `ally.js` which is used by `@next/react-dev-overlay`.

Believe it or not, I'm actually working on migrating from Gatsby 😅 so there a few things that could be broken in my build process so maybe my build isn't outputting something it should. I'll try to dig deeper.

---

I tried downgrading to 9.3 and that actually got it partially working again.

**Reply by PJijin** [reply]
Cool, I have downgraded to 9.3. 😄

**Comment by chrisvariety** [comment]
I'm seeing this as well with a fresh checkout of https://github.com/vercel/next.js/tree/4e5e3b9072ad1775d709da6927dc54aacb13fe9a/examples/with-route-as-modal

Node: v12.17.0
Mac OS X 10.15.4

git checkout, then...
```
yarn
yarn dev
```

Visit http://localhost:3000/ & open up console to see error.

I couldn't get it to reproduce on https://codesandbox.io/, it may be a mac-only thing, so linking to a reproduction is tricky.

If you really want to use 9.4, a total hack is to replace `node_modules/ally.js/util/platform.js` with https://gist.github.com/chrisvariety/2429354130eac4f6bf364d68e3722a64, which is basically just https://github.com/bestiejs/platform.js/blob/master/platform.js copied into ally.js directly instead of imported, as something goes wrong here: https://github.com/medialize/ally.js/blob/master/src/util/platform.js#L7. This works great & the error goes away, but would love to see an official fix if possible. If it were me, I'd remove the dependency on `ally.js` entirely, the project doesn't look well updated and platform sniffing like this isn't a great practice.

**Reply by ricokahler** [reply]
I’ve also found this out and I agree 100%. I think react-trap-focus would be a much better alternative and it’s 10x smaller than ally.js: https://github.com/davidtheclark/focus-trap-react

**Reply by ricokahler** [reply]
I have an experimental PR for this https://github.com/vercel/next.js/pull/14372

**Reply by JoeyBG** [reply]
Any chance you can get back to this? I am experiencing the same issue! :)

**Comment by chrisvariety** [comment, marked as answer]
Looks like there's a potential fix here: https://github.com/vercel/next.js/discussions/13777

I do have `NODE_PATH` set to `.` globally, so it seems likely related to that.

**Reply by PJijin** [reply]
Yeah. This solves the white page error. The page is shown along with the popup error.

```
Unhandled Runtime Error
SyntaxError: Unexpected token < in JSON at position 0
```

**Comment by najmuddinkaishar93** [comment]
NODE_PATH='.'
You will get the reported error in the browser console but nothing in the server log.

The fix for me is to change the NODE_PATH to '/'

// .env file at the root of the project

NODE_PATH='/'
